raiwidgets/tests/test_responsibleai_dashboard_input.py

```diff
@@ -220,3 +222,44 @@ def test_rai_dashboard_input_housing_causal_whatif_failure(
         assert "Failed to generate causal what-if," in \
             flask_server_prediction_output[
                 WidgetRequestResponseConstants.error]
+
+    def test_rai_dashboard_input_housing_causal_causal_global_effects_success(
+            self, create_rai_insights_object_regression):
+        ri = create_rai_insights_object_regression
+        id = ri.causal.get()[0].id
+        causal_whatif_test_data = ri.test.head(1).drop(
+            "target", axis=1).to_dict(orient='records')
+
+        dashboard_input = ResponsibleAIDashboardInput(ri)
+        post_data = (id, causal_whatif_test_data)
+        flask_server_prediction_output = dashboard_input.causal_global_effects(
+            post_data)
+
+        assert flask_server_prediction_output is not None
+        assert WidgetRequestResponseConstants.data in \
+            flask_server_prediction_output
+        assert WidgetRequestResponseConstants.error not in \
+            flask_server_prediction_output
+
+    def test_rai_dashboard_input_housing_causal_causal_global_effects_failure(
+            self, create_rai_insights_object_regression):
+        ri = create_rai_insights_object_regression
+        id = "some_id_that_does_not_exist"
+        causal_whatif_test_data = ri.test.head(1).drop(
+            "target", axis=1).to_dict(orient='records')
+
+        dashboard_input = ResponsibleAIDashboardInput(ri)
+        post_data = (id, causal_whatif_test_data)
+        flask_server_prediction_output = dashboard_input.causal_global_effects(
```

Comment:
nit: I think most of the code in this test from lines 245-254 and in the test above from lines 228-236 could be refactored into a common helper method (and id passed as a parameter) to reduce code duplication

Reply:
Refactored the tests into per feature file.